src/gluonts/distribution/beta.py

```diff
@@ -69,14 +69,20 @@ def log_prob(self, x: Tensor) -> Tensor:
         F = self.F
         alpha, beta = self.alpha, self.beta
 
-        return (
-            (alpha - 1) * F.log(x)
-            + (beta - 1) * F.log(1 - x)
+        # add machine epsilon to avoid infinities due to 0 padding which is masked,
+        # however, causes issues in backpropagation
+        epsilon = np.finfo(self._dtype).eps
```

Comment:
Since you defined `dtype` as a `@property`, why not using that instead?

Reply:
you mean calling that function?

Reply:
No need to call it, since it’s a @property you can just use self.dtype

Reply:
ok